File: 数据结构入门/Contains Duplicate/containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
// var containsDuplicate = function (nums) {
//     return new Set(nums).size !== nums.length;
// };

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
    const _nums = nums.sort();
    for (let i = 0; i < _nums.length; i++) {
        if (_nums[i] === _nums[i + 1]) return true;
    }
    return false;
};

const nums = [1, 2, 3, 4];
console.log(containsDuplicate(nums));

File: 数据结构入门/Contains Duplicate/containsDuplicate.go
package main

import "fmt"

func containsDuplicate(nums []int) bool {
	var customMap = make(map[int]bool)
	for _, v := range nums {
		if customMap[v] {
			return true
		}
		customMap[v] = true
	}
	return false
}

func main() {
	nums := [...]int{1, 2, 3, 1}
	result := containsDuplicate(nums[:])
	fmt.Println(result)
}
